src/content/ja/2022/javascript.md

```diff
@@ -0,0 +1,1074 @@
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=1640758017&format=interactive",
+    sheets_gid="943216000",
+    sql_file="unused_js_bytes_distribution.sql"
+  )
+}}
+
+<a hreflang="en" href="https://web.dev/unused-javascript/">Lighthouse</a> によると、中央のモバイル ページでは、162KBの未使用のJavaScriptがロードされているそうです。90パーセンタイルでは604KBのJavaScriptが、未使用であることがわかります。これは、未使用のJavaScriptの中央値と90パーセンタイルがそれぞれ155KBと598KBであった昨年からわずかに上昇したものです。とくにこの分析ではJavaScriptリソースの転送サイズを追跡しているため圧縮されている場合は、使用されたJavaScriptの解凍部分が、このグラフが示すよりもはるかに大きい可能性があることを考えると、この量は未使用のJavaScriptに相当します。
+
+モバイルページで読み込まれた総バイト数の中央値と比較すると、未使用のJavaScriptは、読み込まれたスクリプト全体の35%を占めています。これは、昨年の36%からわずかに減少していますが、それでも未使用で読み込まれるバイトの大部分であることに変わりはありません。このことは多くのページが、現在のページでは使われないかもしれない、あるいはページのライフサイクルの後半でのインタラクションによって引き起こされるスクリプトを読み込んでおり、起動コストを減らすために[ダイナミックな`import()`](#ダイナミックなimport) が有効かもしれないことを示唆しています。
+
+## 1ページあたりのJavaScriptリクエスト数
+
+ページ上のすべてのリソースは、少なくとも1つのリクエストを開始し、リソースがより多くのリソースに対して追加のリクエストを行う場合は、より多くのリクエストを開始する可能性があります。
+
+スクリプトのリクエストは多ければ多いほど、より多くのJavaScriptを読み込むだけでなくスクリプトリソース間の競合が増え、メインスレッドを停滞させ起動が、遅くなる可能性が高くなるのです。
+
+{{ figure_markup(
+    image="requests-per-page.png",
+    caption="1ページあたりのJavaScriptリクエスト数の分布。",
+    description="1ページあたりのJavaScriptリクエストの10、25、50、75、90パーセンタイルを示す棒グラフ。モバイルページでは、4、10、21、37、60の値になっています。デスクトップでは、全体的に1～3リクエストとやや高い傾向にあります。",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=372493183&format=interactive",
+    sheets_gid="1148191446",
+    sql_file="requests_2022.sql"
+  )
+}}
+
+2022年、モバイルページの中央値は21のJavaScriptリクエストに応答したのに対し、90パーセンタイルでは60のリクエストに応答しています。昨年と比較すると、中央値で1リクエスト、90パーセンタイルで4リクエストの増加です。
+
+2022年のデスクトップデバイスでは、JavaScriptのリクエスト数が中央値で22、90パーセンタイルで63になっています。昨年と比較すると、中央値で1件、90パーセンタイルで4件増加しており、モバイルデバイスと同じ増加率になっています。
+
+リクエスト数の増加幅は大きくないものの、2019年のWeb Almanac開始以来、前年比でリクエスト増加の傾向が続いています。
+
+## JavaScriptはどのように処理されるのですか？
+
+Node.jsのようなJavaScriptランタイムの出現以来、JavaScriptをバンドルし変換するためビルドツールへ依存することがますます一般的になってきました。これらのツールは紛れもなく便利ですが、JavaScriptの提供量に影響を与える可能性があります。今年のWeb Almanacでは、新たにバンドルとトランスパイラの使用状況に関するデータを提示します。
+
+### バンドル
+
+JavaScriptバンドルは、プロジェクトのJavaScriptソースコードを処理し、変換や最適化を適用するビルドタイムツールです。出力はプロダクションレディのJavaScriptです。たとえば、次のようなコードです。
+
+```js
+function sum (a, b) {
+  return a + b;
+}
+```
+
+バンドルはこのコードをより小さく、しかしより最適化された同等品に変換し、ブラウザがダウンロードする時間をより短くします。
+
+```js
+function n(n,r){return n+r}
+```
+
+バンドルは、ソースコードを最適化し、実稼働環境でのパフォーマンスを向上させるための重要な役割を担っています。
+
+JavaScriptのバンドルに関しては豊富な選択肢がありますが、よく頭に浮かぶのは<a hreflang="en" href="https://webpack.js.org/">webpack</a>でしょう。幸いなことに、webpackの生成するJavaScriptにはいくつかのシグネチャ（たとえば `webpackJsonp`）が含まれており、ウェブサイトのプロダクションJavaScriptがwebpackを使ってバンドルされているかどうかを検出することが可能です。
+
+{{ figure_markup(
+    image="webpack-rank.png",
+    caption="webpack-bundled JavaScriptを使用しているページの順位。",
```

Comment:
Should the "-bundled" bit be translated? This means "JavaScript bundled through webpack"

Reply:
Thanks I translate.